core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -149,7 +150,7 @@ class SharedAllocationRecord<void, void> {
       SharedAllocationRecord* arg_root,
 #endif
       SharedAllocationHeader* arg_alloc_ptr, size_t arg_alloc_size,
-      function_type arg_dealloc);
+      function_type arg_dealloc, const std::string& label);
```

Comment:
Consider taking it by value and moving it.

Reply:
In most (all?) cases, we obtain the label as a `const std::string&` already so I opted for passing by `const std::string&` here, too. If there are other good reasons to rather pass by value and move, I am happy to adapt, of course.

Reply:
As written, there's guaranteed to be a copy into the ultimate `m_label` member.

In the common case where the label is coming from some temporary object, Damien's suggestion will allow the compiler to do elision, and construct the temporary directly into the function argument, which will then get inexpensively moved from.

Reply:
I agree with that but we never pass a temporary object here.
Anyway, if anyone requests changes, I will change.